docs/design/metadata.md

```diff
@@ -0,0 +1,365 @@
+# Metadata formats
+
+## Purpose
+
+This document describes the three metadata formats used for variant
+wheels: `pyproject.toml` tables, distribution metadata fields
+and `variants.json` file. All three formats are related, as they share
+a common part and form a single pipeline:
+
+    pyproject.toml → distribution (wheel) metadata → variants.json
```

Comment:
This somehow implies sdists are left out.

Reply:
Hmm, sdists are represented by `pyproject.toml` here. Are you suggesting we make that more explicit, or just skip "wheel"?

Reply:
> sdists are represented by `pyproject.toml` here.

Ah, that wasn't. They're not quite the same. I think it'd be good to explicitly use `PKG-INFO` (say "does not contain modifications" or "contains the same as `METADATA") and `METADATA` to point at the exact metadata files.

Reply:
Okay, I've clarified on `PKG-INFO` and `METADATA`. However, I'd rather not list `PKG-INFO` as if it were used, given that the current implementation does not use that file at all. And given #44, it may become irrelevant too.

Reply:
SGTM. The current text looks clear, and I agree that after gh-44 the potential for confusion is even less. So I'll resolve this comment.